File: 1996-the-number-of-weak-characters-in-the-game/1996-the-number-of-weak-characters-in-the-game.py
class Solution:
        def numberOfWeakCharacters(self, properties: List[List[int]]) -> int:

            properties.sort(key= lambda x:(x[0],-x[1]))
            st,c = [],0
            for ele in properties:
                while st and st[-1][0]<ele[0] and st[-1][1]<ele[1]:
                    st.pop()
                    c+=1
                st.append(ele)

            return c

File: 1996-the-number-of-weak-characters-in-the-game/1996-the-number-of-weak-characters-in-the-game.java
class Solution {
    public int numberOfWeakCharacters(int[][] properties) {
        Arrays.sort(properties, (a, b) -> a[0] == b[0] ? b[1] - a[1] : a[0] - b[0]);
        int max = 0, res = 0;
        for (int i = properties.length - 1; i >= 0; i--) {
            if (properties[i][1] < max) res++;
            max = Math.max(max, properties[i][1]);
        }
        return res;
    }
}